reactive_graph/src/signal/mapped.rs

```diff
@@ -0,0 +1,358 @@
+    }
+}
+
+impl<T> Write for ArcMappedSignal<T>
+where
+    T: 'static,
+{
+    type Value = T;
+
+    fn try_write_untracked(
+        &self,
+    ) -> Option<impl DerefMut<Target = Self::Value>> {
+        let mut guard = self.try_write()?;
+        guard.untrack();
+        Some(guard)
+    }
+
+    fn try_write(&self) -> Option<impl UntrackableGuard<Target = Self::Value>> {
+        let inner = (self.try_write)()?;
+        let inner = DoubleDeref { inner };
+        Some(inner)
+    }
+}
+
+/// A wrapper for a smart pointer that implements [`Deref`] and [`DerefMut`]
+/// by dereferencing the type *inside* the smart pointer.
+///
+/// This is quite obscure and mostly useful for situations in which we want
+/// a wrapper for `Box<dyn Deref<Target = T>>` that dereferences to `T` rather
+/// than dereferencing to `dyn Deref<Target = T>`.
+///
+/// This is used internally in [`MappedSignal`] and [`ArcMappedSignal`].
+pub struct DoubleDeref<T> {
+    inner: T,
+}
+
+impl<T> Deref for DoubleDeref<T>
+where
+    T: Deref,
+    T::Target: Deref,
+{
+    type Target = <T::Target as Deref>::Target;
+
+    fn deref(&self) -> &Self::Target {
+        self.inner.deref().deref()
+    }
+}
+
+impl<T> DerefMut for DoubleDeref<T>
+where
+    T: DerefMut,
+    T::Target: DerefMut,
+{
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        self.inner.deref_mut().deref_mut()
+    }
+}
+
+impl<T> UntrackableGuard for DoubleDeref<T>
+where
+    T: UntrackableGuard,
+    T::Target: DerefMut,
+{
+    fn untrack(&mut self) {
+        self.inner.untrack();
+    }
+}
+
+/// A derived signal type that wraps an [`RwSignal`] with a mapping function,
+///  allowing you to read or write directly to one of its field.
+///
+/// Tracking the mapped signal tracks changes to *any* part of the signal, and updating the signal notifies
+/// and notifies *all* depenendencies of the signal. This is not a mechanism for fine-grained reactive updates
+/// to more complex data structures. Instead, it allows you to provide a signal-like API for wrapped types
+/// without exposing the original type directly to users.
+pub struct MappedSignal<T, S = SyncStorage> {
+    #[cfg(debug_assertions)]
+    defined_at: &'static Location<'static>,
+    inner: StoredValue<ArcMappedSignal<T>, S>,
+}
+
+impl<T> MappedSignal<T> {
+    /// Wraps a signal with the given mapping functions for shared and exclusive references.
+    #[track_caller]
+    pub fn new<U>(
+        inner: RwSignal<U>,
+        map: fn(&U) -> &T,
+        map_mut: fn(&mut U) -> &mut T,
+    ) -> Self
+    where
+        T: Send + Sync + 'static,
+        U: Send + Sync + 'static,
+    {
+        Self {
+            #[cfg(debug_assertions)]
+            defined_at: Location::caller(),
+            inner: {
+                let this = ArcRwSignal::from(inner);
+                StoredValue::new_with_storage(ArcMappedSignal::new(
+                    this, map, map_mut,
+                ))
+            },
+        }
+    }
+}
+
```

Comment:
```suggestion
impl<T> Copy for MappedSignal<T> {}

impl<T> Clone for MappedSignal<T> {
    fn clone(&self) -> Self {
        *self
    }
}
```

Reply:
Thanks! Not at all too late, will just make a separate PR. Easy to fix.

Reply:
#3827